### ISSUE TITLE
Docs: document no-restricted-syntax support for mixed warn + error configuration

### ISSUE THREAD

**Issue by NateRadebaugh** [eslint/eslint#20291]
### Docs page(s)

Right now it appears that the no-restricted-syntax rule must be all "warn" or all "error" but I don't see a documented way to determine the severity on a per-rule basis.

For example, I'd like to mark the first as a warning and the second as a lint error, but it is unclear using the documentation how to achieve this goal.

```ts
"no-restricted-syntax": [
  "warn",
  {
    selector: "NewExpression[callee.name='Date']",
    message: "Avoid using 'new Date'.",
  },

  // Disallow default export for things named *Service
  {
    selector: "ExportDefaultDeclaration > Identifier[name=/Service$/]",
    message:
      "Avoid using default exports for services. Please use named exports instead.",
  },
],
```

LLMs hallucinate a `"severity"` property

### What documentation issue do you want to solve?

https://eslint.org/docs/latest/rules/no-restricted-syntax

### What do you think is the correct solution?

A simple example of both warn and error rule in the same configuration would be helpful.

### Participation

- [x] I am willing to submit a pull request for this change.

### Additional comments

Is this even possible?

**Comment by DMartens** [comment]
Unfortunately setting the severity per selector is currently not possible as the rule has no way to set the severity for specific reports. Only the user can set the rule severity in the configuration ([related issue](https://github.com/eslint/eslint/issues/19463)).
As a workaround for `no-restricted-syntax` you can create a "plugin" in your flat `eslint.config.js`:
```javascript
function createSyntaxRule({ selector, message }) {
	return {
		create(context) {
			return {
				[selector](node) {
					context.report({ message, node });
				}
			};
		}
	}
}

export default {
	// ... other configuration
 	plugins: {
		own: {
			rules: {
				'no-new-date': createSyntaxRule({
					selector: "NewExpression[callee.name='Date']",
					message: "Avoid using 'new Date'.",
				}),
				// your other rules
			}
		}
	},
	rules: {
		'own/no-new-date': 'error', // As these are now separate rules, the severity can be set differently
	}
}
```

This also allows to disable the rule for specific files via an override.

**Comment by kirkwaiblinger** [comment]
A workaround idea I've seen in the wild is to create a utility for creating a different rule per selector so that they can be separately configured, see https://github.com/bradzacher/eslint-no-restricted

**Comment by dohrisalim** [comment]
I can add documentation for the no-restricted-syntax rule's capability to mix severity levels. I understand that this powerful feature isn't well-documented, and I'll create clear examples showing how to configure different selectors with warn vs error severity in the same rule configuration.

**Comment by JoshuaKGoldberg** [comment]
This use case isn't specific to `no-restricted-syntax`. It really applies to all general-purpose restriction rules.

* [`@typescript-eslint/no-restricted-imports`](https://typescript-eslint.io/rules/no-restricted-imports) / [`no-restricted-imports`](https://eslint.org/docs/latest/rules/no-restricted-imports)
* [`@typescript-eslint/no-restricted-types`](https://typescript-eslint.io/rules/no-restricted-types)
* [`no-restricted-exports`](https://eslint.org/docs/latest/rules/no-restricted-exports)
* [`no-restricted-globals`](https://eslint.org/docs/latest/rules/no-restricted-globals)
* [`no-restricted-properties`](https://eslint.org/docs/latest/rules/no-restricted-properties)
* [`no-restricted-syntax`](https://eslint.org/docs/latest/rules/no-restricted-syntax)

I'm personally overly biased against warnings, as I think [bulk suppressions](https://eslint.org/docs/latest/use/suppressions) should be the preferred way to gradually enable rules over time. So I'm default -1 on taking any strong action on this. But, if other team members think this use case is important to document, I'd be in favor of adding an FAQ and/or example to the rules' docs pages.

**Comment by snitin315** [comment]
I agree with Josh, this applies to all general-purpose restriction rules. The current documentation already looks fine to me, but if we do decide to update it, we should make sure all related rule docs are updated as well.

**Comment by mdjermanovic** [comment]
> Right now it appears that the no-restricted-syntax rule must be all "warn" or all "error" but I don't see a documented way to determine the severity on a per-rule basis.

This is not documented because it is not supported by ESLint core. There are workarounds, though.

I see there are proposals to document how to achieve this for any general-purpose rule (or generally any rule), but I'm not sure what approach would be documented?

I think the most viable workaround that can apply to any rule is to "clone" the rule definition into a custom plugin, as described in https://github.com/eslint/eslint/issues/18242#issuecomment-2026910531. However, even if we decide to document this approach, I think it should wait for https://github.com/eslint/eslint/issues/19013 because core rules are currently not exported aside from the `use-at-your-own-risk` entry point, which I think should not be mentioned in the docs.
